### PR TITLE
Feat: cloudflare https container reg civo creds

### PR DESCRIPTION
**Pull request by claywd**
* Makes ECR optional, defaults to git provider 
* Enables Cloudflare as a dns provider
* Actions now default to https when communicating with the git providers, while gitops operations default to ssh and can be overrided to use https with an additional parameter. 
* Refactors AWS bootstrapping to reuse code
* Refactors multiple pieces of code to reduce duplication where possible and conform AWS to the same scheme as the other cloud providers
* Uses latest runtime code

### CODE REVIEW

**Review by jarededwards — APPROVED** [review]

File: internal/controller/controller.go [hunk 1]
```diff
@@ -30,7 +31,7 @@ import (
 )
 
 const (
-	gitopsTemplateVersion = "v2.2.0"
+	gitopsTemplateVersion = "main"
```
Comment (CristhianF7) [inline internal/controller/controller.go]: I think we need to pin this

**Review by jarededwards — APPROVED** [review]

**Review by CristhianF7 — DISMISSED** [review]

File: internal/controller/controller.go (hunk 1 above)
Comment (claywd) [reply internal/controller/controller.go]: you are correct.

Comment (claywd) [reply internal/controller/controller.go]: pinned to 2.2.7

**Review by CristhianF7 — APPROVED** [review]
Great work 🚀

**Pull request merged**